### PR TITLE
[docs] Add call tracing howto doc page

### PR DESCRIPTION
**Pull request by andreilee**
Signed-off-by: Andrei Lee <andreilee@fb.com>

## Summary

Added a basic call tracing howto page on Magma docusaurus.

calltracing_screenshot

## Test Plan

N/A

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/docusaurus/sidebars.json [hunk 1]
```diff
@@ -35,15 +35,27 @@
...
+          "nms/nms_dashboard",
+          "nms/nms_equipment",
+          "nms/nms_network",
+          "nms/nms_subscriber",
+          "nms/nms_traffic",
+          "nms/nms_organizations",
+          "nms/nms_how_to_guide"
+        ]
+      },
+      {
```
Comment (karthiksubraveti) [inline docs/docusaurus/sidebars.json:55]: Let's move this out of NMS user guide and call this NMS developer guide and add more pages under this.

File: docs/docusaurus/sidebars.json [hunk 2]
```diff
@@ -65,7 +77,12 @@
         "ids": ["lte/tr069", "lte/s1ap_tests", "lte/dev_notes"]
       }
     ],
-    "Feature Guides": ["howtos/ue_metering", "howtos/config_agw_bridged", "howtos/thanos"],
+    "Feature Guides": [
+      "howtos/ue_metering",
+      "howtos/config_agw_bridged",
+      "howtos/thanos",
+      "howtos/call_tracing"
```
Comment (karthiksubraveti) [inline docs/docusaurus/sidebars.json:87]: I'm trying to understand the right place we should add this guide. I feel the right fit is NMS, but will probably broadcast it in the team and get some broader consensus here.

Comment (karthiksubraveti) [reply docs/docusaurus/sidebars.json:87]: Alternatively, if you can add the ### Initiating a Call Trace and ### Viewing a call trace to NMS user guide also that will be great. Thanks.

Comment (andreilee) [reply docs/docusaurus/sidebars.json:87]: I think I'm gonna make a separate PR for these larger changes to re-organize the docs

File: docs/readmes/howtos/call_tracing.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Ensure you have the following:
+
+* a functional orc8r
+* a configured LTE network
+* a configured LTE gateway with eNodeB
+* subscribers that can attach to your LTE gateway
+* network running in un-federated mode
+
+### Initiating a Call Trace
```
Comment (karthiksubraveti) [inline docs/readmes/howtos/call_tracing.md:30]: For the sake of completion, can you add a API reference section and add all the relevant APIs which can be used to control call tracing as well

File: docs/readmes/howtos/call_tracing.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+Once the call trace is started, you can stop and/or download the call trace
+on the same page. Under `Actions`, click the vertical ellipsis button for your
+call trace to see these options.
+
+### Viewing a Call Trace
+
+It is suggested that Wireshark is used to analyze call trace captures.
+
+![Call Tracing Actions](assets/nms/calltracing_actions.png)
+
```
Comment (karthiksubraveti) [inline docs/readmes/howtos/call_tracing.md:57]: Can you add a debugging section mentioning how to debug issues if call tracing is not working as expected. Where to look, logs to collect etc.

File: docs/readmes/howtos/call_tracing.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+# Call Tracing
+
+Magma supports basic traffic capture for troubleshooting purposes.
+Monitoring of control messaging flow and other traffic between the Magma access
+gateway and eNodeB devices is possible with this feature.
+
+As of the time of writing, we are in the process of adding filtering for
+specific protocols and allowing custom options through tshark.
+
```
Comment (karthiksubraveti) [inline docs/readmes/howtos/call_tracing.md:17]: Can you add the max size of call tracing buffer we currently support and any other assumptions here.

File: docs/readmes/howtos/call_tracing.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+![Call Tracing Actions](assets/nms/calltracing_actions.png)
+
+### Additional Configuration
+
+To do additional configuration for call traces, modify the `ctraced.yml` file
+on the access gateway.
+
+This allows you to configure the following options:
+
+- network interfaces to capture traffic on
```
Comment (karthiksubraveti) [inline docs/readmes/howtos/call_tracing.md:65]: Can you specify exact config knobs which can be used for customizing these 
- Can you add examples for each of the option on how one should go about adding the network interfaces to capture traffic on
- How to modify to modify trace files stored on the access gateway
- different tools which can be used for capturing traffic.

File: docs/readmes/howtos/call_tracing.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+![Call Tracing Page](assets/nms/calltracing_page.png)
+
+The call tracing page can be accessed directly via the NMS sidebar.
+
+Click `Start New Trace`
+
+![Call Tracing Dialog](assets/nms/calltracing_dialog.png)
+
+`trace_id` must be a unique value among all the call traces on your network.
```
Comment (karthiksubraveti) [inline docs/readmes/howtos/call_tracing.md:40]: Just curious, do we throw an error in UI itself, if we specify an already existing call trace ID ?

File: docs/readmes/howtos/call_tracing.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+capturing across multiple gateways simultaneously are not supported at this
+time. If you wish to do so, start multiple call traces, one for each gateway.
+
+Once the call trace is started, you can stop and/or download the call trace
+on the same page. Under `Actions`, click the vertical ellipsis button for your
+call trace to see these options.
+
+### Viewing a Call Trace
+
+It is suggested that Wireshark is used to analyze call trace captures.
```
Comment (karthiksubraveti) [inline docs/readmes/howtos/call_tracing.md:54]: Can you share a screenshot where a downloaded pcapng file is opened in wireshark and we can filter it further based on protocol or IMSI ?

**Review by karthiksubraveti — DISMISSED** [review]
Thanks @andreilee for adding this doc. Making a separate PR for covering the top level sections sounds good. Added a few comments here.

File: docs/readmes/howtos/call_tracing.md (hunk 7 above)
Comment (andreilee) [reply docs/readmes/howtos/call_tracing.md:40]: Error is thrown in the UI

**Comment by rdefosse** [comment]
Can one of the admins verify this patch?

**Review by karthiksubraveti — APPROVED** [review]

**Pull request merged**